Netcracker/KubeMarine
# [CPDEV-96115]-Enabling_strong_crypto_support_by_default

### Description
These cipher suites are configuring the server to support a selection of strong and secure cryptographic algorithms for secure communication over TLS (Transport Layer Security).
* 

**Fixes # (issue)**
[CPDEV-96115](https://tms.netcracker.com/browse/CPDEV-96115)

**Security Strength:**
These cipher suites prioritize the use of strong cryptographic algorithms, enhancing the security strength of the TLS connections. They provide forward secrecy through the ECDHE key exchange.

**Cipher Suite Diversity:**
Including a mix of AES-GCM and ChaCha20-Poly1305 cipher suites ensures compatibility with a broader range of clients and provides alternatives in case certain algorithms are vulnerable or less secure.

### Solution

Enhanced Security:
The configured cipher suites enhance the security of the TLS communication by using modern cryptographic algorithms and key exchange methods.

Forward Secrecy:
Ephemeral key exchange (ECDHE) ensures forward secrecy, meaning that even if the server's private key is compromised, past communications remain secure.

Compatibility:
The chosen cipher suites strike a balance between security and compatibility. 

Performance:
ChaCha20-Poly1305 cipher suites are known for their good performance, which can be beneficial in resource-constrained environments.
* 


### Test Cases

**TestCase 1**

Test Configuration:

- OS: Ubuntu-20.04

Steps:

1. Installed a miniha cluster
2. Installed kube-bench to make KubeMarine CIS aligned. 
3. Run scan using command - kube-bench and check for identifier **1.2.31**.

Results:

| Before | After |
| ------ | ------ |
| Result used to be Warning | Result shows Pass |


**TestCase 2**
Apply the patch for the old clusters where API Server Cryptographic Ciphers are not update even after even after k8s upgrade.

Test Configuration:
Steps: 
1. Try to upgrade k8s for e.g. from v1.24.11 to v1.25.2 
2. With such upgrade --tls-cipher-suites on /etc/kubernetes/manifests/kube-apiserver.yaml will not get updated.
3. Then apply this patch runing kubemarine migrate_kubemarine command.
4. Check the /etc/kubernetes/manifests/kube-apiserver.yaml file to verify if --tls-cipher-suites is updated and also  run scan using command - kube-bench and check for identifier **1.2.31**.

Results:

| Before | After |
| ------ | ------ |
| Result used to be Warning | Result shows Pass |


**TestCase 3**
Apply the patch for the old clusters where API Server Cryptographic Ciphers are already available or some other ciphers are already present.

Results:
It will skip the patch by logging ciphers are already present.

### Checklist
